### PR TITLE
midi module for macOs

### PR DESCRIPTION
**Pull request by psmokotnin**
Fixed midi module for macOS in MU4

### CODE REVIEW

File: framework/midi/internal/platform/osx/coremidiinport.h [hunk 1]
```diff
@@ -22,6 +22,8 @@
 #include <memory>
 #include "imidiinport.h"
 
+class MIDIPacketList;
```
Comment (igorkorsukov) [inline framework/midi/internal/platform/osx/coremidiinport.h:25]: unnecessary, you need to remove

File: framework/midi/internal/platform/osx/coremidiinport.h [hunk 2]
```diff
@@ -42,15 +44,17 @@ class CoreMidiInPort : public IMidiInPort
...
     void doProcess(uint32_t message, tick_t timing);
 
+private:
     struct Core;
     std::unique_ptr<Core> m_core;
     std::string m_deviceID;
     bool m_running = false;
     async::Channel<std::pair<tick_t, Event> > m_eventReceived;
+
+    void initCore();
```
Comment (igorkorsukov) [inline framework/midi/internal/platform/osx/coremidiinport.h:57]: better methods first, then class members

File: framework/midi/internal/midiportdatasender.cpp [hunk 3]
```diff
@@ -53,8 +53,6 @@ bool MidiPortDataSender::sendEvents(tick_t fromTick, tick_t toTick)
         const Event& event = pos->second;
 
         if (SKIP_EVENTS.find(event.type) == SKIP_EVENTS.end()) {
-            // noop
```
Comment (Jojo-Schmitz) [inline framework/midi/internal/midiportdatasender.cpp:56]: Shouldn't that `==` turn into a `!=` then?

Comment (psmokotnin) [reply framework/midi/internal/midiportdatasender.cpp:56]: It was a bug. Now it works as it should: send event only if it is not in the skip list.
```
if (SKIP_EVENTS.find(event.type) == SKIP_EVENTS.end()) {
    midiOutPort()->sendEvent(event);
}
```

File: framework/midi/internal/platform/osx/coremidiinport.h (hunk 2 above)
Comment (psmokotnin) [reply framework/midi/internal/platform/osx/coremidiinport.h]: fixed

File: framework/midi/internal/platform/osx/coremidiinport.h (hunk 1 above)
Comment (psmokotnin) [reply framework/midi/internal/platform/osx/coremidiinport.h]: fixed

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**